packages/core-app-api/src/app/App.test.tsx

```diff
@@ -322,6 +338,68 @@ describe('Integration Test', () => {
     });
   });
 
+  it('should track route changes via analytics api', async () => {
+    const mockAnalyticsApi = new MockAnalyticsApi();
+    const apis = [createApiFactory(analyticsApiRef, mockAnalyticsApi)];
+    const app = new PrivateAppImpl({
+      apis,
+      defaultApis: [],
+      themes: [
+        {
+          id: 'light',
+          title: 'Light Theme',
+          variant: 'light',
+          theme: lightTheme,
+        },
+      ],
+      icons: defaultAppIcons,
+      plugins: [],
+      components,
+      bindRoutes: ({ bind }) => {
+        bind(plugin1.externalRoutes, {
+          extRouteRef1: plugin1RouteRef,
+          extRouteRef2: plugin2RouteRef,
+        });
+      },
+    });
+
+    const Provider = app.getProvider();
+    const Router = app.getRouter();
```

Comment:
Do you think providing a wrapper would reduce the boilerplate?
For example:

**file.test.tsx**
```jsx
it('should track route changes via analytics api', () => {
  const { analyticsApi } = renderWithAnalyticsApi(
     <Routes>
        <Route path="/" element={<NavigateComponent />} />
        <Route path="/foo" element={<HiddenComponent path="/foo" />} />
      </Routes>
  );
  expect(analyticsApi).toHaveBeenCapturedEvent({ ... }); // custom Jest Matcher
});
```

Reply:
This would definitely make this test way clearer, but I'm not sure it makes sense to break it out for just this one test case.  Perhaps if I studied the rest of the test cases in this suite, I could extract a generic `renderInApp()` where one could provide arbitrary...  API implementations?  Plugins?  Componnets?  Bound routes?  Etc.  Then it might make sense.

Arguably the `renderInTestApp()` function in `test-utils` should be able to do this?  ...But that might be a task unto itself.